src/latex/latex_matcher.py

```diff
@@ -42,14 +42,22 @@ def run(self):
         for filepath in filepaths:
             i += 1
             logger.info(f"Working on `{filepath}` - {i}/{total}")
-
-            arxiv_id = re.search(ARXIV_ID_REGEX, filepath)[1]
+            match = re.search(ARXIV_ID_REGEX, filepath)
+            if not match:
+                continue
+            arxiv_id = match[1]
             results.setdefault(arxiv_id, {})
             with open(filepath) as fp:
                 text = fp.read()
-
-                repos_ids = repos_finder.find(arxiv_id, text, contextualized=True)
-                results[arxiv_id] = pub_finder.find(arxiv_id, repos_ids)
+                if not (repos_ids := repos_finder.find(arxiv_id, text, contextualized=False)):
+                    logging.error(f"latex_matcher: No repo ids found in {arxiv_id}")
```

Comment:
FIY, the prefix `latex_matcher` is not needed, because the log line will contain the name of the logger "Latex Matcher" (line 12):
```suggestion
                    logging.error(f"No repo ids found in {arxiv_id}")
```